parser/prism/Parser.cc

```diff
@@ -79,4 +91,119 @@ vector<core::LocOffsets> Parser::collectCommentLocations() {
 
     return commentLocations;
 }
+
+bool Parser::isTUntyped(pm_node_t *node) const {
+    if (!node || !PM_NODE_TYPE_P(node, PM_CALL_NODE)) {
+        return false;
+    }
+
+    pm_call_node_t *call = down_cast<pm_call_node_t>(node);
+    if (!call->receiver || PM_NODE_TYPE_P(call->receiver, PM_CONSTANT_PATH_NODE)) {
+        return false;
+    }
+
+    pm_constant_path_node_t *receiver = down_cast<pm_constant_path_node_t>(call->receiver);
+    if (receiver->parent != nullptr) {
+        return false; // Should be root-anchored ::T
+    }
+
+    auto methodName = resolveConstant(call->name);
+    if (methodName != "untyped"sv) {
+        return false;
+    }
+
+    auto receiverName = resolveConstant(receiver->name);
+    return receiverName == "T"sv;
```

Comment:
For these constant you _could_ do something like:

```suggestion
    return receiverName == core::Names::T.().shortName(gs); 
```

But it's slower, and IMO, less clear.